src/test/java/io/stargate/sgv2/jsonapi/api/v1/VectorSearchIntegrationTest.java

```diff
@@ -1315,6 +1316,123 @@ public void deleteOne() {
           .body("status", is(nullValue()))
           .body("errors", is(nullValue()));
     }
+
+    @Test
+    @Order(8)
+    public void insertVectorWithUnmatchedSize() {
+      createVectorCollection(namespaceName, vectorSizeTestCollectionName, 5);
+      // Insert data with $vector array size less than vector index defined size.
+      final String vectorStrCount3 = buildVectorElements(0, 3);
+      String jsonVectorStrCount3 =
+          """
+                      {
+                         "insertOne": {
+                            "document": {
+                                "_id": "shortHandedVectorElements",
+                                "name": "shortHandedVectorElements",
+                                "description": "this document should have vector size as 5",
+                                "$vector": [ %s ]
+                            }
+                         }
+                      }
+                      """
+              .formatted(vectorStrCount3);
+
+      given()
+          .header(HttpConstants.AUTHENTICATION_TOKEN_HEADER_NAME, getAuthToken())
+          .contentType(ContentType.JSON)
+          .body(jsonVectorStrCount3)
+          .when()
+          .post(CollectionResource.BASE_PATH, namespaceName, vectorSizeTestCollectionName)
+          .then()
+          .statusCode(200)
+          .body("errors", is(notNullValue()))
+          .body("errors[0].message", endsWith("Expected vector of 5 size, but received 3"));
```

Comment:
maybe better to say "expected vector of length 5", "5 size" reads a bit awkwardly.

Reply:
Thanks! Fixed that.